src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubBuildStatusNotification.java

```diff
@@ -182,32 +190,45 @@ private static GitHubSCMSource getSCMSource(final SCMSourceOwner scmSourceOwner)
-                            String url = DisplayURLProvider.get().getJobURL(job);
-                            // Has not been built yet, so we can only guess that the current PR head is what will be built.
-                            // In fact the submitter might push another commit before this build even starts.
-                            createCommitStatus(repo, pr.getHead().getSha(), GHCommitState.PENDING, url, Messages.GitHubBuildStatusNotification_CommitStatus_Queued(), head);
-                        }
-                    } catch (FileNotFoundException fnfe) {
-                        LOGGER.log(Level.WARNING, "Could not update commit status to PENDING. Valid scan credentials? Valid scopes?");
-                        LOGGER.log(Level.FINE, null, fnfe);
-                    } catch (IOException ioe) {
-                        LOGGER.log(Level.WARNING, "Could not update commit status to PENDING. Message: " + ioe.getMessage());
-                        LOGGER.log(Level.FINE, null, ioe);
-                    }
-                }
-            }
+         @Override public void onEnterWaiting(Queue.WaitingItem wi) {
+             if (wi.task instanceof Job) {
+                 Job<?,?> job = (Job) wi.task;
+                 // TODO would actually be better to use GitHubSCMSource.retrieve(SCMHead, TaskListener)   (when properly implemented).
+                 // That would allow us to find the current head commit even for non-PR jobs.
+                 SCMHead head = SCMHead.HeadByItem.findHead(job);
+                 ItemGroup<?> multiBranchProject = job.getParent();
+                 GitHubSCMSource source = null;
+                 if (multiBranchProject instanceof SCMSourceOwner) {
+                   SCMSourceOwner scmSourceOwner = (SCMSourceOwner) multiBranchProject;
+                   source = getSCMSource(scmSourceOwner);
+                 }
+                 if (source!=null) {
+                     if (head instanceof PullRequestSCMHead) {
+                         try {
+                             GHRepository repo = lookUpRepo(job);
+                             if (repo != null) {
+                                 int number = ((PullRequestSCMHead) head).getNumber();
+                                 GHPullRequest pr = repo.getPullRequest(number);
+                                 String url = DisplayURLProvider.get().getJobURL(job);
+                                 try {
+                                     url = job.getAbsoluteUrl();
+                                 } catch (IllegalStateException ise) {
+                                     url = "http://unconfigured-jenkins-location/" + job.getUrl();
+                                 }
+                                 // Has not been built yet, so we can only guess that the current PR head  is what will be built.
+                                 // In fact the submitter might push another commit before this build      even starts.
```

Comment:
something is wrong with the spacing here and the lines below. Maybe a tab -> whitespace mis-transformation?